src/core/core.js

```diff
@@ -457,28 +463,27 @@ var p5 = function(sketch, node, sync) {
           f.call(self);
         }
       });
-
-      // remove window bound properties and methods
-      if (this._isGlobal) {
-        for (var p in p5.prototype) {
+    }
+    // remove window bound properties and methods
+    if (this._isGlobal) {
+      for (var p in p5.prototype) {
+        try {
+          delete window[p];
+        } catch (x) {
+          window[p] = undefined;
+        }
+      }
+      for (var p2 in this) {
+        if (this.hasOwnProperty(p2)) {
           try {
-            delete window[p];
+            delete window[p2];
           } catch (x) {
-            window[p] = undefined;
-          }
-        }
-        for (var p2 in this) {
-          if (this.hasOwnProperty(p2)) {
-            try {
-              delete window[p2];
-            } catch (x) {
-              window[p2] = undefined;
-            }
+            window[p2] = undefined;
           }
         }
       }
+      p5.instance = null;
```

Comment:
can you explain this line here, I'm not sure I understand what this does?

Reply:
This removes the `p5.instance` reference to sketch from hanging around in global scope after removal. I guess it isn't necessary addition but it lets the GC cleanup more than otherwise.

Reply:
ah got it, i missed the place where it was assigned. makes sense.